domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -343,6 +426,11 @@ class ExplorationProgressController @Inject constructor(
           "Cannot navigate to a next state if an answer submission is pending."
         }
         explorationProgress.stateDeck.navigateToNextState()
+
+        // only mark checkpoint if current state is pending state. This ensures that checkpoint will
+        // not be marked on any of the completed states.
+        if (explorationProgress.stateDeck.isCurrentStateTopOfDeck())
+          saveExplorationCheckpoint()
```

Comment:
Separate point, but please make sure to add braces for single-line cases like this. We only omit braces if the entire if + statement fits on the same line. See: https://developer.android.com/kotlin/style-guide#braces.

Reply:
done

Reply:
indentation seems incorrect

Reply:
indentation is auto-formated.